Dockerfile

```diff
@@ -0,0 +1,15 @@
+# Dockerfile to perform a full cluster-operator build from source.
+#
+# This Dockerfile is intended for performing in-cluster OpenShift builds. It differs
+# from those in build/ in that it does not expect to compile the binary externally
+# and add to the container, everything is built internally during docker build.
+FROM golang:1.9
+
+ENV PATH=/go/bin:$PATH GOPATH=/go
+
+ADD . /go/src/github.com/openshift/cluster-operator
+
+WORKDIR /go/src/github.com/openshift/cluster-operator
+RUN NO_DOCKER=1 make build
+ENTRYPOINT ["bin/cluster-operator"]
+
```

Comment:
Other Dockerfiles are not suitable for OpenShift in-cluster builds, there's a lot of layers of Docker build, multi-arch stuff, and inherently we build the binary outside and add to container. Adding this small new Dockerfile for openshift builds was a simple workaround.

Reply:
We should use our regular Dockerfile to build the images so that they match what would be built by hand or elsewhere. This can be accomplished by using this Dockerfile to build the binaries and push the resulting image to an imagestream. Then use the image as a source with the regular Dockerfile. https://docs.openshift.org/latest/dev_guide/builds/build_inputs.html#image-source